source/gui/settingsDialogs.py

```diff
@@ -990,9 +990,9 @@ def makeSettings(self, settingsSizer):
 			if globalVars.appArgs.secure:
 				mirrorBox.Disable()
 
-	def onChangeMirrorURL(self, evt):
+	def onChangeMirrorURL(self, evt: wx.Event):
```

Comment:
Here's how I usually check these types: https://docs.wxpython.org/wx.Button.html#events-events-emitted-by-this-class

```suggestion
	def onChangeMirrorURL(self, evt: wx.CommandEvent):
```

Reply:
This is technically incorrect, as `GeneralSettingsPanel.onChangeMirrorURL` is called in response to activation of the "Change..." button, and in response to the enter key being pressed (via `GeneralSettingsPanel._enterTriggersOnChangeMirrorURL`). Since the latter case receives and so forwards a `wx.KeyEvent`, I used the closest common superclass of `wx.KeyEvent` and `wx.CommandEvent`, which is `wx.Event`. I've narrowed the type hint to be the intersection of the two subclasses it receives though.

Reply:
Ah ok thanks for explaining